src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -559,6 +556,33 @@ private SmugglerProgressBase.Counts ProcessDocuments(SmugglerResult result, Buil
             return result.Documents;
         }
 
+        private void AddInfoToSmugglerResult(SmugglerResult result, string message)
+        {
+            result.AddInfo(message);
+            _onProgress.Invoke(result.Progress);
+        }
+
+        private void SetNonPersistentFlagsIfNeeded(BuildVersionType buildType, DocumentItem item, SmugglerResult result)
+        {
+            if (_options.SkipRevisionCreation)
+            {
+                item.Document.NonPersistentFlags |= NonPersistentDocumentFlags.SkipRevisionCreation;
+            }
+
+            if (buildType == BuildVersionType.V3 && _options.OperateOnTypes.HasFlag(DatabaseItemType.RevisionDocuments) == false)
+            {
+                item.Document.NonPersistentFlags |= NonPersistentDocumentFlags.SkipLegacyRevision;
+
+                if ((item.Document.NonPersistentFlags & NonPersistentDocumentFlags.LegacyRevision) == NonPersistentDocumentFlags.LegacyRevision &&
```

Comment:
This should be moved to a method 'SkipDocument' or something like that. Side-effect and flow worries me. We are increasing the skipped count, adding a flag and passing that to 'WriteDocuments' so it can be skipped inside.

```csharp
SetNonPersistentFlagsIfNeeded(buildType, item);

if (SkipDocument(item))
{
    result.Documents.SkippedCount++;
    continue;
}
```

Reply:
done